### DISCUSSION TITLE
How to exclude outdated package version in generated imports?

### DISCUSSION THREAD

**Discussion by ErikKalkoken** [golang/vscode-go#3613] [Help]
When I use a new package the import statements for that package are added automatically, which is a great and helpful feature.

However, when there is a v2 version of that package the plugin still always adds the v1 version. This can lead to errors by accidentally using an outdated package version in your code or at least always always requires me to manually update the import statement to the newer version.

I there a way to exclude outdated versions in generated imports?

**Comment by ousloob** [comment]
@ErikKalkoken 
You can add this in your `settings.json`

```json
"go.useLanguageServer": true,
"gopls": {
    "allowModfileModifications": true,
    "allowImplicitNetworkAccess": true
}
```

This makes gopls automatically fix outdated imports when you run go mod tidy in VS Code. After that you execute:

```sh
go get -u ./...
go mod tidy
``` 
With that, all dependencies are updated to the latest stable versions.

And last :

```sh
gopls fix ./...
```

If imports are still outdated, force VS Code to reformat them.

**Reply by mihaitodor** [reply]
> `"allowImplicitNetworkAccess": true`

Looks like this setting [was deprecated](https://github.com/golang/go/issues/66860) recently.

**Reply by ousloob** [reply]
Oh true.
If we use `go get -u ./...`, we don't need `allowImplicitNetworkAccess`

If we want to make dependency updates easier, we can add `go mod tidy` to a Makefile, so we don’t have to run it manually every time. For example:

```makefile
tidy:
    go get -u ./...
    go mod tidy
```	
This can be useful if we have a lot of dependencies to manage.

Personally, I also like to keep an eye on updates to avoid unexpected breaking changes, but automating it can save time in larger projects.

**Comment by ErikKalkoken** [comment]
I tried this, but I get an "invalid option" error for `"allowModfileModifications"`.

Do I maybe need a special version?I am running Go extension v0.44.0 with Go 1.23.4.

**Reply by ousloob** [reply]
@ErikKalkoken indeed it's deprecated as well.
The only solution is the command I shared with `go get`

**Reply by ErikKalkoken** [reply]
That is new information. So far you guys only said that `allowImplicitNetworkAccess` is deprecated.

So, with both configuration options being deprecated, what is the solution then? Just running getting all updates and running mod tidy does not solve it.

And btw: `gopls fix ./...` is deprecated too apparently.

**Reply by ousloob** [reply]
Yes, `allowImplicitNetworkAccess` is [deprecated ](https://github.com/golang/go/issues/66860) in last april.

And I just realised that `go get -u ./...` updates only the minor and fix, since the major versions often introduce breaking changes.

I personnally encourage you to update the major manually so you can be aware about the breaking changes.

However, if you're unable to, I haven't tested it myself, but this [package](https://github.com/icholy/gomajor) might be a possible solution.
